service-registrar/src/main/java/com/msm/sr/service/status/notification/type/NotificationType.java

```diff
@@ -1,9 +1,24 @@
 package com.msm.sr.service.status.notification.type;
 
+import com.msm.sr.service.status.notification.notifier.ServiceStatusNotifier;
+
 /**
  * @author riste.jovanoski
  * @since 6/19/2017
  */
 public enum NotificationType {
-    REGISTRATION, DEREGISTRATION
+    REGISTRATION {
+        @Override
+        public void notify(ServiceStatusNotifier serviceStatusNotifier, String... parameters) {
+            serviceStatusNotifier.notifyForRegistration(parameters);
+        }
+    },
+    DEREGISTRATION {
+        @Override
+        public void notify(ServiceStatusNotifier serviceStatusNotifier, String... parameters) {
+            serviceStatusNotifier.notifyForDeregistration(parameters);
+        }
+    };
+
+    public abstract void notify(ServiceStatusNotifier serviceStatusNotifier, String... parameters);
```

Comment:
Avoid using `notify` as a method name because it collides with the provides `notify` method used for waking up threads.